iOS/DoraemonKit/Src/Core/Plugin/Performance/MotionMonitor/Data/DoraemonMotionDataSource.m

```diff
@@ -0,0 +1,62 @@
+//
+//  DoraemonMotionDataSource.m
+//  CocoaAsyncSocket
+//
+//  Created by tianYang on 2022/11/4.
+//
+
+#import "DoraemonMotionDataSource.h"
+
+@implementation DoraemonMotionDataSource {
+    dispatch_semaphore_t semaphore;
+}
++ (DoraemonMotionDataSource *)shareInstance{
+    static dispatch_once_t once;
+    static DoraemonMotionDataSource *instance;
+    dispatch_once(&once, ^{
+        instance = [[DoraemonMotionDataSource alloc] init];
+    });
+    return instance;
+}
+
+- (instancetype)init{
+    self = [super init];
+    if (self) {
+        _motionUseModelArray = [NSMutableArray array];
+        semaphore = dispatch_semaphore_create(1);
+    }
+    return self;
+}
+
+- (void)addUseModel:(DoraemonMotionDataModel *)useModel {
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
+    [_motionUseModelArray insertObject:useModel atIndex:0];
+    dispatch_semaphore_signal(semaphore);
+}
+
+- (void)clear{
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
+    [_motionUseModelArray removeAllObjects];
+    dispatch_semaphore_signal(semaphore);
+}
+
+- (NSString *)toJson {
+    NSMutableArray *dicArray = @[].mutableCopy;
```

Comment:
why not surrounded by semaphore here?

Reply:
We use the same naming method as the other singletons in Dokit
<img width="465" alt="image" src="https://user-images.githubusercontent.com/19932964/210086419-44c3fe05-d606-49bf-9d3d-04d3b4a61dcf.png">

Reply:
in here，we just read the _motionUseModelArray